src/depends/NAT/nat.cpp

```diff
@@ -0,0 +1,169 @@
+#include "nat.h"
+#include "libUtils/Logger.h"
+#include <arpa/inet.h> //INET6_ADDRSTRLEN
+#include <cstring>
+#include <iostream>
+#include <miniupnpc/miniupnpc.h>
+#include <miniupnpc/upnpcommands.h>
+
+using namespace std;
+
+NAT::NAT()
+{
+    m_urls = make_shared<UPNPUrls>();
+    m_data = make_shared<IGDdatas>();
+
+    memset(m_urls.get(), 0, sizeof(struct UPNPUrls));
+    memset(m_data.get(), 0, sizeof(struct IGDdatas));
+
+    m_initialized = false;
+
+    struct UPNPDev* devlist;
+    int error = 0;
+
+#if MINIUPNPC_API_VERSION >= 14
+    devlist = upnpDiscover(2000, NULL, NULL, 0, 0, 2, &error);
```

Comment:
What are all these magic numbers? Can we refactor it.. Can be just a variable in this class

Reply:
A good practice is to declare these as const variables in an anonymous namespace inside the .cpp e.g.

```
// foo.cpp
namepsace {
    const int MY_CONSTANT = 1;
} // close anonymous namespace
```

> If you plan to expose them to other classes then they have to be static members.